File: ai/src/main/java/org/pingel/gestalt/core/TransformEdge.java
package org.pingel.gestalt.core;

import org.pingel.util.LabelledDirectedEdge;

public class TransformEdge extends LabelledDirectedEdge<TransformVertex>
{
    public Name transformName;
    public Traversal traversal;
    
    public TransformEdge(Name transformName,
    		Traversal traversal,
			TransformVertex source,
			TransformVertex dest)
    {
        super(source, transformName.base, dest);
        this.transformName = transformName;
    	this.traversal = traversal;
    }

    
    public String toString() {
    	
    	if( traversal == null ) {
    		return "apply " + transformName + " " +
    		getSource().toString() + " " + getDest().toString();
    	}
    	else {
    		return "apply " + transformName + " " +
    		getSource().toString() + "." + traversal + " " + getDest().toString();
    	}
    }
}

File: ai/src/main/scala/org/pingel/gestalt/core/TransformEdge.scala
package org.pingel.gestalt.core

import org.pingel.util.LabelledDirectedEdge

case class TransformEdge(transformName: Name, traversal: Traversal, source: TransformVertex, dest: TransformVertex)
extends LabelledDirectedEdge[TransformVertex]
{
    def getSource() = source
    
    def getDest() = dest
    
    def getLabel() = transformName.base
    
    def toString(): String = {
      (traversal == null) match {
        case true  => "apply " + transformName + " " + source.toString() + " " + dest.toString()
        case false => "apply " + transformName + " " + source.toString() + "." + traversal + " " + dest.toString()
      }
    }
    
}
